e2e/testenv/fail.go

```diff
@@ -0,0 +1,35 @@
+package testenv
+
+import (
+	"os"
+	"os/exec"
+	"path"
+
+	ginkgo "github.com/onsi/ginkgo/v2"
+)
+
+func FailAndGather(message string, callerSkip ...int) {
+	if _, err := exec.LookPath("crust-gather"); err != nil {
+		ginkgo.GinkgoWriter.Print("⛔ crust-gather is not available, not dumping cluster info")
+		ginkgo.Fail(message, callerSkip...)
+	}
+
+	pwd := os.Getenv("GITHUB_WORKSPACE")
+	if pwd != "" {
+		pwd = path.Join(pwd, "tmp", "upstream")
+	} else {
+		pwd = path.Join(os.TempDir(), "fleet-gather")
+	}
+	path := path.Join(pwd, ginkgo.CurrentSpecReport().FullText())
+
+	ginkgo.GinkgoWriter.Printf("   Gathering cluster info for '%s' to '%s'...\n", ginkgo.CurrentSpecReport().FullText(), pwd)
```

Comment:
I see squares here, presumably a fonts issue; what are these chars meant to represent? Perhaps we could use more common characters?

Reply:
These are little bugs...